src/Gui/Splashscreen.cpp

```diff
@@ -493,343 +540,397 @@ QString AboutDialog::getAdditionalLicenseInformation() const
+    licenseView->setAttribute(Qt::WA_DeleteOnClose);
+    licenseView->show();
+    QString title = tr("License");
+    QString fragment = link.fragment();
+    if (fragment.startsWith(QLatin1String("_Toc"))) {
+        QString prefix = fragment.mid(4);
+        title = QString::fromLatin1("%1 %2").arg(prefix, title);
+    }
+    licenseView->setWindowTitle(title);
+    getMainWindow()->addWindow(licenseView);
+    licenseView->setSource(link);
+}
+
+
+/**
+ * Having got this far in the refactoring (mainly SRP), it is clear that this
+ * function is very much the tail wagging the dog, ie demanding info from all over the app,
+ * violating "Tell, Don't Ask".
+ *
+ * It has far too many responsibilities, and far too many dependencies.
+ *
+ * Without a working system this function is impossible to test.
+ *
+ * At the end of the day, this function is just combining disparate information strings. And that's
+ * all it should be doing.
+ *
+ * The config items should be put in config. But currently config is just a variable, so it
+ * should be wrapped in a class with various io handlers (value in/out as stdString, QString
+ * etc), itemExists() function, valueEmpty() functionn etc.
+*/
+std::string AboutDialog::makeSummaryReport()
+{
+
+    // should be moved to where config is managed
+    // at this level we should not need to know the internals of config
+    auto configFind = [](std::string name) {
+        std::map<std::string, std::string>& config = App::Application::Config();
+        auto it = config.find(name);
+        return it == config.end() ? "" : it->second;
     };
 
-    // Coin3D
-    libInfo << LibraryInfo {
-        QLatin1String("Coin3D"),
-        baseurl + QLatin1String("#_TocCoin3D"),
-        QLatin1String("https://coin3d.github.io"),
-        QLatin1String(COIN_VERSION)
+    // should be moved to where config is managed
+    // at this level we should not need to know the internals of config
+    auto configFetch = [](std::string name) {
+        return App::Application::Config()[name];
     };
 
-    // Eigen3
-    libInfo << LibraryInfo {
-        QLatin1String("Eigen"),
-        baseurl + QLatin1String("#_TocEigen"),
-        QLatin1String("https://eigen.tuxfamily.org"),
-        QString::fromLatin1(FC_EIGEN3_VERSION)
+    auto sysEnv = [](const std::string & name) {
+        return QProcessEnvironment::systemEnvironment()
+            .value(QString::fromStdString(name), QString())
+            .toStdString();
     };
 
-    // FreeType
-    libInfo << LibraryInfo {
-        QLatin1String("FreeType"),
-        baseurl + QLatin1String("#_TocFreeType"),
-        QLatin1String("https://freetype.org"),
-        QString::fromLatin1(FC_FREETYPE_VERSION)
+    auto deskInfo = [&sysEnv] {
+        std::string deskEnv = sysEnv("XDG_CURRENT_DESKTOP");
+        std::string deskSess = sysEnv("DESKTOP_SESSION");
+
+        std::string result {};
+        if (!(deskEnv.empty() && deskSess.empty())) {

```

Comment:
According to the cited [Clean Code Rules](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29) negative conditionals should be avoided. Instead: `if (!deskEnv.empty() || !deskSess.empty()) {`